library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3Reconciler.java

```diff
@@ -0,0 +1,364 @@
+      S3Reconciler.OutputMetaData metaData = doneTuples.poll();
+      logger.info("found metaData = {}", metaData);
+      committedTuples.remove(metaData);
+      try {
+        Path dest = new Path(metaData.getPath());
+        //Deleting the intermediate files and when writing to tmp files
+        // there can be vagrant tmp files which we have to clean
+        FileStatus[] statuses = fs.listStatus(dest.getParent());
+        logger.info("found statuses = {}", statuses);
+
+        for (FileStatus status : statuses) {
+          String statusName = status.getPath().getName();
+          logger.info("statusName = {}", statusName);
+          if (statusName.endsWith(TMP_EXTENSION) && statusName.startsWith(metaData.getFileName())) {
+            //a tmp file has tmp extension always preceded by timestamp
+            String actualFileName = statusName.substring(0,
+                statusName.lastIndexOf('.', statusName.lastIndexOf('.') - 1));
+            logger.info("actualFileName = {}", actualFileName);
+            if (metaData.getFileName().equals(actualFileName)) {
+              logger.info("deleting stray file {}", statusName);
+              fs.delete(status.getPath(), true);
+            }
+          } else if (statusName.equals(metaData.getFileName())) {
+            logger.info("deleting s3-compaction file {}", statusName);
+            fs.delete(status.getPath(), true);
+          }
+        }
+      } catch (IOException e) {
+        logger.error("Unable to Delete a file: {}", metaData.getFileName());
+      }
+    }
+  }
+
+  /**
+   * Get access key id
+   *
+   * @return Access key id for Amazon S3
+   */
+  public String getAccessKey()
+  {
+    return accessKey;
+  }
+
+  /**
+   * Set access key id
+   *
+   * @param accessKey
+   *          Access key id for Amazon S3
+   */
+  public void setAccessKey(String accessKey)
+  {
+    this.accessKey = accessKey;
+  }
+
+  /**
+   * Get secret key
+   *
+   * @return Secret key for Amazon S3
+   */
+  public String getSecretKey()
+  {
+    return secretKey;
+  }
+
+  /**
+   * Set secret key
+   *
+   * @param secretKey
+   *          Secret key for Amazon S3
+   */
+  public void setSecretKey(String secretKey)
+  {
+    this.secretKey = secretKey;
+  }
+
+  /**
+   * Get bucket name
+   *
+   * @return Bucket name for data upload
+   */
+  public String getBucketName()
+  {
+    return bucketName;
+  }
+
+  /**
+   * Set bucket name
+   *
+   * @param bucketName
+   *          Bucket name for data upload
+   */
+  public void setBucketName(String bucketName)
+  {
+    this.bucketName = bucketName;
+  }
+
+  /**
+   * Get directory name
+   *
+   * @return Directory name under S3 bucket
+   */
+  public String getDirectoryName()
+  {
+    return directoryName;
+  }
+
+  /**
+   * Set directory name
+   *
+   * @param directoryName
+   *          Directory name under S3 bucket
+   */
+  public void setDirectoryName(String directoryName)
+  {
```

Comment:
Can you please add a check @NotNull for secretkey, directoryName, bucketName setters.